xchange-stream-coinmate/src/main/java/info/bitrich/xchangestream/coinmate/CoinmateStreamingService.java

```diff
@@ -0,0 +1,53 @@
+package info.bitrich.xchangestream.coinmate;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import info.bitrich.xchangestream.coinmate.dto.auth.AuthParams;
+import info.bitrich.xchangestream.service.netty.JsonNettyStreamingService;
+import io.reactivex.Observable;
+import io.reactivex.subjects.PublishSubject;
+import sun.rmi.runtime.Log;
+
+import java.io.IOException;
+
+/**
+ * The new API uses one WebSocket connection per channel, unlike other exchanges,
+ * which use channels over one WebSocket. This is a hack to use JsonNettyStreamingService as
+ * single channel source.
+ */
+public class CoinmateStreamingService extends JsonNettyStreamingService {
+
+  private PublishSubject<JsonNode> messages;
+
+  CoinmateStreamingService(String url) {
+    super(url);
+    this.messages = PublishSubject.create();
+  }
+
+  @Override
+  protected String getChannelNameFromMessage(JsonNode message) throws IOException {
+    return null;
```

Comment:
I think this should return whatever the channel this service represents (so pass in the channel name as a constructor argument).  That way `subscribeChannel` will work normally.

Reply:
You are right. I changed it back to `subscribeChannel` and it works now.